README.rst

```diff
@@ -28,51 +23,36 @@
+(gateway-auth).
 
-Quickstart
-----------
 
-In order to get an instance of Renku up and running, clone the main Renku
+Developing the gateway-auth component
+-------------------------------------
+The renku gateway-auth component is best developped in the context of a full renku 
+deployment. In order to get an instance of Renku up and running, clone the main Renku
 repository and follow these instructions_.
 
 .. _instructions: https://renku.readthedocs.io/en/latest/developer/setup.html
 
-Developing the gateway
-----------------------
-Once you have an instance of Renku running locally, you could modify the gateway code
-and restart the platform through the :code:`make minikube-deploy` command to see the
-changes. However, this will make for a very poor development experience as the deployment
-process is optimized for production.
-
-Instead we recommend connecting to your minikube (or any other kubernetes cluster) through
-telepresence_. Once telepresence is installed, create a python environment and install
-the necessary python dependencies by running :code:`pipenv install`. Then, start a
-telepresence shell through :code:`make dev` and launch a development server by executing
-the prompted command inside the telepresence shell.
-
-.. _telepresence: https://www.telepresence.io/reference/install
+Once you have an instance of Renku running, you could modify the gateway code, build the 
+image, re-build the chart, redeploy, etc... This will make for a poor development experience
+with very long feedback cycles.
 
-The gateway in development setting is now available under the ip-address of your
-minikube cluster (:code:`${minikube ip}/api`) and you should see requests from the
-Renku UI appear in the logs.
+Instead we recommend intercepting traffic to the gateway-auth component and routing it to
+your local machine through telepresence_ (note that currently you MUST use version 2.4.X, 
+mac users see in particular tele-troubleshooting_). Once telepresence is installed, create a 
+python environment and install the necessary python dependencies by running 
+:code:`pipenv install --dev`. Then, create a telepresence intercept using the dedicated 
+:code:`./telepresence-intercept.sh` script and follow the instructions. This will forward 
+all requests to the gateway-auth service deployed in the cluster to a flask development 
+server running on your local machine (with hot reloading, etc). You can now use your 
+favourite IDE and develop the component completely locally. 
 
-So what is happening here? The command :code:`make dev` launches telepresence which
-swaps the renku-gateway service in your minikube deployment for a locally running version of
-the gateway served by a flask development server. This gives you live updates on code change
-in a minikube deployment!
+.. _telepresence: https://www.telepresence.io/docs/v2.4/quick-start/
+.. _tele-troubleshooting: https://www.telepresence.io/docs/latest/troubleshooting/
```

Comment:
Here somewhere or maybe even better as a message from the scripts used to launch a telepresence sessions we should put some info on how to clean things up.

I.e. until you run `telepresence leave XXX` your Renku deployment does not go back to normal.

Also it seems that XXX = <service name>-<namespace> but we should double check.

Reply:
If you provide a command with the intercept, the intercept will be terminated as soon as that command exits. I added a comment about this to the readme.